pkg/controllers/customeradmin/group_controller.go

```diff
@@ -0,0 +1,132 @@
+package customeradmin
+
+import (
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	userv1client "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	osaCustomerAdmins  = "osa-customer-admins"
+	osaCustomerReaders = "osa-customer-readers"
+)
+
+type reconcileGroup struct {
+	osclient  *openshiftClient
+	aadClient *graphrbac.GroupsClient
+	aadConfig map[string]string
+	log       *logrus.Entry
+	groupMap  map[string]string
+	Admins    string `yaml:"admins,omitempty"`
```

Comment:
this'll go into a separate config structure - when it does, use `json:"...`, not `yaml:"...`

Reply:
ok